csharp/ql/src/semmle/code/csharp/Generics.qll

```diff
@@ -196,19 +201,24 @@ class TypeParameter extends DotNet::TypeParameter, Type, @type_parameter {
  * ```
  */
 class TypeParameterConstraints extends Element, @type_parameter_constraints {
-  /** Gets a specific interface constraint (if any). */
+  /** Gets a specific interface constraint, if any. */
   Interface getAnInterfaceConstraint() {
     specific_type_parameter_constraints(this, getTypeRef(result))
   }
 
-  /** Gets a specific type parameter constraint (if any). */
+  /** Gets a specific type parameter constraint, if any. */
   TypeParameter getATypeParameterConstraint() {
     specific_type_parameter_constraints(this, getTypeRef(result))
   }
 
-  /** Gets the specific class constraint (if any). */
+  /** Gets the specific class constraint, if any. */
   Class getClassConstraint() { specific_type_parameter_constraints(this, getTypeRef(result)) }
 
+  /** Gets an annotated specific type constraint, if any. */
+  AnnotatedType getAnAnnotatedTypeConstraint() {
```

Comment:
I would somehow expect predicates `getAnAnnotatedInterfaceConstraint()`, `getAnAnnotatedTypeParameterConstraint()`, and `getAnAnnotatedClassConstraint()` instead.

Reply:
Or perhaps we should deprecate `getInterfaceConstraint()`, `getTypeParameterConstraint()`, and `getAClassConstraint()`, and replace with just a single `getTypeConstraint()`?